packages/ddp-client/test/livedata_tests.js

```diff
@@ -414,11 +414,7 @@ testAsyncMulti('livedata - compound methods', [
       );
     } catch (e) {
     }
-    if (Meteor.isClient)
-      // client can fool itself by cheating, but only until the sync
-      // finishes
-      await checkBalances(test, -10, 160);
-    else await checkBalances(test, 90, 60);
```

Comment:
@zodern maybe from this test, the api should be stub first, then the server result? because now it needs to await for the server result.
Not sure if this is correct

Reply:
If you look at the old code for the test: 
https://github.com/meteor/meteor/blob/e0763170e98d05b48b00ec47d5a03c8f2a00c09e/packages/ddp-client/test/livedata_tests.js#L340-L398

You could maybe implement it for Meteor 3 like this:

```js
let promise = Meteor.callAsync(
  'ledger/transfer',
  test.runId(),
  'alice',
  'bob',
  100,
  true,
);

if (Meteor.isClient) {
  // client can fool itself by cheating, but only until the sync
  // finishes
  await promise.stub
  await checkBalances(test, -10, 160);
}

await promise.catch(err => {
  failure(test, 409)(err, result);
});

await checkBalances(test, 90, 60);

```

Reply:
I think I made a mistake somewhere in here https://github.com/meteor/meteor/blob/fd2150c4c2d41a6a625b54cf6bad41598f8a6176/packages/ddp-client/common/livedata_connection.js#L891-L907

The test still does not pass

Reply:
I took at look at this over the weekend. Unfortunately wasn't able to make the test pass. I noticed that `checkBalances` on the client is running before the stub promise has completed. When I added a timeout of 1, prior to `checkBalances` it passed. 

```js
if (Meteor.isClient) {
  // client can fool itself by cheating, but only until the sync
  // finishes
  await promise.stub // this isn't being awaited
  // if i add -- await sleep(1) -- it passes
  await checkBalances(test, -10, 160);
}
```

Hope this can be of help. Maybe someone closer to the codebase will be able to instantly tell what the issue is. 🙂 If I get some time this week, I will take a closer look.